eth_test_parser/src/revm_builder/env.rs

```diff
@@ -96,8 +116,14 @@ impl TestBody {
                     data: tx_shared_data.data[m.indexes.data].clone(),
                     chain_id: Some(MATIC_CHAIN_ID),
                     nonce: self.transaction.nonce.try_into().ok(),
-                    // TODO: Add `access_list` to `Transaction` and use it here.
-                    access_list: vec![],
+                    // `access_list` is defined parallel to `transaction.data` in the test filler
+                    // definitions.
+                    // https://ethereum-tests.readthedocs.io/en/latest/test_filler/test_transaction_state.html?highlight=access#fields
+                    access_list: tx_shared_data
+                        .access_list
+                        .get(m.indexes.data)
+                        .cloned()
+                        .unwrap_or_default(),
```

Comment:
I think ideally we would panic if an index is out of range, so we know to check if there's a bug on our side or the test is bad.

Is the issue here that `indexes.data` may be 0 when no access lists are given? I guess ideally we'd default to `[[]]`, but if that's nontrivial then I think this is fine.